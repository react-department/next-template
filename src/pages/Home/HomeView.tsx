import { useTranslation } from 'next-i18next';
import Layout from '../../layout/Layout';
import Title from '../../components/Title/Title';
import Cards from './components/Cards';
import styles from './sass/Home.module.scss';

function HomeView() {
  const { t } = useTranslation('home');

  return (
    <Layout>
      <div className={styles.page}>
        <Title>
          {t('welcomeMessage')}
          {' '}
          <a href="https://nextjs.org">
            {t('nextJs')}
            !
          </a>
        </Title>
        <p className={styles.description}>
          {t('getStarted')}
          {' '}
          <code className={styles.code}>pages/index.js</code>
        </p>
        <Cards />
      </div>
    </Layout>
  );
}

export default HomeView;
