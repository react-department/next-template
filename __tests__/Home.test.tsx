import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from '../pages/index';

describe('Home', () => {
  it('renders a heading', () => {
    render(<Home />);

    const heading = screen.getByRole('heading', {
      name: /welcomeMessage/i,
    });

    expect(heading).toBeInTheDocument();
  });
  it('renders homepage unchanged', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
