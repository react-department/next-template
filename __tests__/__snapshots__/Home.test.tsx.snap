// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div
    className="container"
  >
    <main
      className="main"
    >
      <div
        className="page"
      >
        <h1
          className="title"
        >
          welcomeMessage
           
          <a
            href="https://nextjs.org"
          >
            nextJs
            !
          </a>
        </h1>
        <p
          className="description"
        >
          getStarted
           
          <code
            className="code"
          >
            pages/index.js
          </code>
        </p>
        <div
          className="grid"
        >
          <a
            className="card"
            href="https://nextjs.org/docs"
          >
            <h3>
              documentation
               →
            </h3>
            <p>
              findInformation
            </p>
          </a>
          <a
            className="card"
            href="https://nextjs.org/learn"
          >
            <h3>
              learn
               →
            </h3>
            <p>
              learnAbout
            </p>
          </a>
          <a
            className="card"
            href="https://github.com/vercel/next.js/tree/master/examples"
          >
            <h3>
              examples
               →
            </h3>
            <p>
              discoverAndDeploy
            </p>
          </a>
        </div>
      </div>
    </main>
    <footer
      className="footer"
    >
      <a
        href="https://onix-systems.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        poweredBy
         
        onix
      </a>
    </footer>
  </div>
</div>
`;
